test/direnv-test-common.sh

```diff
@@ -201,6 +203,21 @@ test_start "failure"
   test_neq "${DIRENV_WATCHES:-}" ""
 test_stop
 
+test_start "watch-dir"
+    echo "No watches by default"
+    test_eq "${DIRENV_WATCHES}" "${WATCHES}"
+
+    direnv_eval
+
+    if ! direnv show_dump "${DIRENV_WATCHES}" | grep -q "testfile"; then
+        echo "FAILED: testfile not added to DIRENV_WATCHES"
+        exit 1
+    fi
```

Comment:
I wasn't sure about adding this here, is it worth it?

Reply:
Yeah, sure! Thanks for adding tests.